internal/update/update.go

```diff
@@ -182,37 +183,62 @@ func latestHomebrewRelease(ctx context.Context, channel string) (*Release, error
 	}, nil
 }
 
+var errBrewNotFound = errors.New("command 'brew' not found")
+
+// memoized value so we're not executing homebrew commands on every call
+// Use brewBinDir()
+var _brewBinDir *string
+
+func brewBinDir() (string, error) {
+	if _brewBinDir != nil {
+		return *_brewBinDir, nil
+	}
+
+	brewExe, err := safeexec.LookPath("brew")
+	if err != nil {
+		return "", errBrewNotFound
+	}
+
+	brewPrefixBytes, err := exec.Command(brewExe, "--prefix").Output()
+	if err != nil {
+		return "", err
+	}
+
+	brewBinPrefix := filepath.Join(strings.TrimSpace(string(brewPrefixBytes)), "bin") + string(filepath.Separator)
+
+	_brewBinDir = api.StringPointer(brewBinPrefix)
+	return *_brewBinDir, nil
```

Comment:
you lost me there :) 

why returning the dereference of the pointer to BrewBinPrefix and not brewBinPrefix directly?

Reply:
ahhh.. because of the global variable to memoize the result 🤔